File: test_en_savoir_plus.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.common.exceptions import TimeoutException

def run_test_en_savoir_plus():
    driver = webdriver.Chrome()
    driver.maximize_window()

    try:
        driver.get("https://connect.algoan.com/v2/init?client_id=030d0c7dfcfdcfcc135c6cf5&redirect_uri=https://dashboard.algoan.com")
        print("Test du bouton principal 'En savoir plus'...")

        try:
            button = WebDriverWait(driver, 10).until(
                EC.element_to_be_clickable((By.XPATH, "//div[contains(text(), 'En savoir plus')]"))
            )
            button.click()
            WebDriverWait(driver, 10).until(
                EC.url_contains("welcome-page-details")
            )
            print("✅ Bouton principal 'En savoir plus' fonctionnel.")
        except TimeoutException:
            print("❌ Test échoué : le bouton principal 'En savoir plus' ou la redirection a échoué.")
    except Exception as e:
        print(f"Erreur inattendue : {e}")
    finally:
        driver.quit()

File: test_en_savoir_plus.ts
import { chromium } from 'playwright';

export async function runTestEnSavoirPlus() {
    const browser = await chromium.launch({ headless: false });
    const context = await browser.newContext();
    const page = await context.newPage();

    try {
        // Charger la page principale
        await page.goto("https://connect.algoan.com/v2/init?client_id=030d0c7dfcfdcfcc135c6cf5&redirect_uri=https://dashboard.algoan.com");
        console.log("Test du bouton principal 'En savoir plus'...");
        await page.waitForTimeout(2000);

        // Localiser et cliquer sur le bouton 'En savoir plus' au milieu de la page
        const button = await page.waitForSelector("//div[contains(text(), 'En savoir plus') and not(ancestor::*[@aria-label='Sidebar'])]", { timeout: 10000 });
        await page.waitForTimeout(2000);
        await button.click();
        console.log("✅ Bouton 'En savoir plus' (au milieu de la page) cliqué.");
        await page.waitForTimeout(2000);

        // Attendre que l'URL contienne "welcome-page-details"
        await page.waitForURL(/welcome-page-details/, { timeout: 10000 });
        await page.waitForTimeout(2000);
        console.log("✅ Redirection vers 'En savoir plus' réussie.");
    } catch (error) {
        console.error("❌ Test échoué : le bouton 'En savoir plus' ou la redirection a échoué.", error);
    } finally {
        await browser.close();
    }
}

if (require.main === module) {
    runTestEnSavoirPlus();
}